File: fix_buttons.py
import os
import re

def walk_dir(directory):
    results = []
    for root, dirs, files in os.walk(directory):
        if 'node_modules' in root or '.next' in root:
            continue
        for file in files:
            if file.endswith('.tsx') or file.endswith('.jsx'):
                results.append(os.path.join(root, file))
    return results

files = walk_dir('./src')

button_regex = re.compile(r'<(Button|button)\b([^>]*)>')

def replace_attrs(match):
    tag = match.group(1)
    attrs = match.group(2)
    
    updated_attrs = re.sub(r'\buppercase\b', 'capitalize', attrs)
    updated_attrs = re.sub(r'\btracking-wide(?:st|r)?\b', '', updated_attrs)
    updated_attrs = re.sub(r'\btext-\[9px\]\b', 'text-[12px]', updated_attrs)
    updated_attrs = re.sub(r'\btext-\[10px\]\b', 'text-[13px]', updated_attrs)
    updated_attrs = re.sub(r'\btext-\[11px\]\b', 'text-[13px]', updated_attrs)
    
    return f'<{tag}{updated_attrs}>'

for filepath in files:
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
        
    original = content
    content = button_regex.sub(replace_attrs, content)
    
    if content != original:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f'Updated: {filepath}')

File: fix_buttons.js
const fs = require('fs');
const path = require('path');

function walkDir(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        const filePath = path.join(dir, file);
        const stat = fs.statSync(filePath);
        // Exclude node_modules and .next etc just in case
        if (stat && stat.isDirectory() && !filePath.includes('node_modules') && !filePath.includes('.next')) {
            results = results.concat(walkDir(filePath));
        } else if (filePath.endsWith('.tsx') || filePath.endsWith('.jsx')) {
            results.push(filePath);
        }
    });
    return results;
}

const files = walkDir('./src');

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    let original = content;

    // We specifically replace uppercase, tracking variants, and tiny font sizes ON buttons
    content = content.replace(/<(Button|button)\b([^>]*)>/g, (match, tag, attrs) => {
        let updatedAttrs = attrs.replace(/\buppercase\b/g, 'capitalize');
        updatedAttrs = updatedAttrs.replace(/\btracking-wide(?:st|r)?\b/g, ''); // Tracking usually looks weird with titlecase
        updatedAttrs = updatedAttrs.replace(/\btext-\[9px\]\b/g, 'text-[12px]');
        updatedAttrs = updatedAttrs.replace(/\btext-\[10px\]\b/g, 'text-[13px]');
        updatedAttrs = updatedAttrs.replace(/\btext-\[11px\]\b/g, 'text-[13px]'); // Keep it balanced
        return `<${tag}${updatedAttrs}>`;
    });

    if (content !== original) {
        fs.writeFileSync(file, content);
        console.log(`Updated: ${file}`);
    }
});
